internal/keygen/keygen.go

```diff
@@ -1,25 +1,91 @@
 package keygen
 
 import (
-	"github.com/vultisig/vultisigner/config"
-	"github.com/vultisig/vultisigner/internal/types"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/vultisig/mobile-tss-lib/coordinator"
+	"github.com/vultisig/mobile-tss-lib/tss"
+	"github.com/vultisig/vultisigner/config"
+	"github.com/vultisig/vultisigner/internal/types"
 )
 
-func JoinKeyGeneration(kg *types.KeyGeneration) error {
+func JoinKeyGeneration(kg *types.KeyGeneration) (string, error) {
+
 	relayServer := config.AppConfig.Relay.Server
+	keyFolder := config.AppConfig.Server.VaultsFilePath
 
-	_, err := coordinator.ExecuteKeyGeneration(coordinator.KeygenInput{
+	// _, err := coordinator.ExecuteKeyGeneration(coordinator.KeygenInput{
+	// 	Server:    relayServer,
+	// 	Key:       kg.Key,
+	// 	Parties:   kg.Parties,
+	// 	Session:   kg.Session,
+	// 	ChainCode: kg.ChainCode,
+	// })
+	// if err != nil {
+	// 	return err
+	// }
+
+	// return nil
+
+	if err := coordinator.RegisterSession(relayServer, kg.Session, kg.Key); err != nil {
```

Comment:
make coordinator part of vultisigner ,  don't rely on mobile-tss-lib to provide coordinate feature